src/test/versionbits_tests.cpp

```diff
@@ -14,6 +14,18 @@
 /* Define a virtual block time, one block per 10 minutes after Nov 14 2014, 0:55:36am */
 static int32_t TestTime(int nHeight) { return 1415926536 + 600 * nHeight; }
 
+static const std::string StateName(ThresholdState state)
+{
+    switch (state) {
+    case ThresholdState::DEFINED:   return "DEFINED";
+    case ThresholdState::STARTED:   return "STARTED";
+    case ThresholdState::LOCKED_IN: return "LOCKED_IN";
+    case ThresholdState::ACTIVE:    return "ACTIVE";
+    case ThresholdState::FAILED:    return "FAILED";
+    } // no default case, so the compiler can warn about missing cases
```

Comment:
nit: Can we add empty `default` statement which does nothing but avoids compiler warning and possible errors in future?

Reply:
That's exactly what the *lack* of default case here does.

If a case is missed, the compiler will warn.

Reply:
This bothered be too. IMO better to use `default`. Enums are bounded. The default state can reasonably be either `ACTIVE` or `FAILED`. This can be future-proofed by using a `static_asset` On the enum, which documents the assumption and catches it at compile time. Warnings aren’t nearly as safe.

Reply:
It's already safe as a missing default case will simply cause the `return ""` after the switch to execute. The only time the return value of this function is used is to print a message if a test case is already failing, so the consequences of having an empty string are trivial -- CI is already failing at that point. Additionally, we bump those missing-case warnings to errors (`-Werror=switch`) when compiling with enable-werror and I think we have that enabled on most CI builds (all except for win64?).

Reply:
I believe we were both referring to an enumeration added in the future being inadvertently missed, in one case caught by a warning and in the other by a halt, not to the correctness of the current behavior.

As for reliance on warnings, we get 55 legit warnings from vc141 from libbitcoinconsensus sources. So warnings can easily be missed.

Reply:
Note that the first two commits of this PR have already been merged, in #21334.